drivers/led_strip/ws2812_uart.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT worldsemi_ws2812_uart
+
+#include <zephyr/drivers/led_strip.h>
+
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ws2812_uart);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/sys/math_extras.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/dt-bindings/led/led.h>
+
+struct ws2812_uart_cfg {
+	const struct device *uart_dev; /* UART device */
+	uint8_t *px_buf;
+	size_t length;
+	uint8_t num_colors;
+	const uint8_t *color_mapping;
+	const uint8_t *uart_lookup_table;
+	uint16_t reset_delay;
+	uint8_t repeat_count;
+};
+
+/*
+ * Serialize a 24-bit color channel value into an equivalent sequence
+ * of 8 bytes of UART frames,.
+ */
+static inline int ws2812_uart_ser(const struct device *dev, uint8_t *buf, uint8_t color[3])
```

Comment:
use serialise/serialize in function name otherwise it looks like it's short for serial